src/core/api/track_choice_manager.ts

```diff
@@ -717,8 +755,43 @@ export default class TrackChoiceManager {
     });
   }
 
-  private _updateAudioTrackChoices() {
-    const preferredAudioTracks = this._preferredAudioTracks.getValue();
+  /**
+   * Reset all audio tracks choices to corresponds to the current preferences.
+   */
+  private _applyAudioPreferences() : void {
+    // Remove all memoized choices and start over
+    this._audioChoiceMemory = new WeakMap();
+    this._resetChosenAudioTracks();
+  }
+
+  /**
+   * Reset all text tracks choices to corresponds to the current preferences.
+   */
+  private _applyTextPreferences() : void {
+    // Remove all memoized choices and start over
```

Comment:
So here, you are really talking about memoization, or did you mean memorization ?

Reply:
You're right, it's more about memorization here.

Reply:
Done